### PR TITLE
Fixes #144 by removing `typing` dep for py3.5+

### LINKED ISSUE #144: typing requirement breaks Python 3.8 and above
When `typing` installed it breaks pip on Python 3.8

The result:
```
  Traceback (most recent call last):
    File "runpy.py", line 193, in _run_module_as_main
      return _run_code(code, main_globals, None,
    File "runpy.py", line 86, in _run_code
      exec(code, run_globals)
    File "pip/__main__.py", line 16, in <module>
      from pip._internal.cli.main import main as _main  # isort:skip # noqa
    File "pip/_internal/cli/main.py", line 10, in <module>
      from pip._internal.cli.autocompletion import autocomplete
    File "pip/_internal/cli/autocompletion.py", line 9, in <module>
      from pip._internal.cli.main_parser import create_main_parser
    File "pip/_internal/cli/main_parser.py", line 7, in <module>
      from pip._internal.cli import cmdoptions
    File "pip/_internal/cli/cmdoptions.py", line 24, in <module>
      from pip._internal.exceptions import CommandError
    File "pip/_internal/exceptions.py", line 10, in <module>
      from pip._vendor.six import iteritems
    File "pip/_vendor/__init__.py", line 62, in <module>
      vendored("cachecontrol")
    File "pip/_vendor/__init__.py", line 33, in vendored
      __import__(modulename, globals(), locals(), level=0)
    File "cachecontrol/__init__.py", line 9, in <module>
      from .wrapper import CacheControl
    File "cachecontrol/wrapper.py", line 1, in <module>
      from .adapter import CacheControlAdapter
    File "cachecontrol/adapter.py", line 5, in <module>
      from requests.adapters import HTTPAdapter
    File "requests/__init__.py", line 112, in <module>
      from . import utils
    File "requests/utils.py", line 25, in <module>
      from . import certs
    File "requests/certs.py", line 15, in <module>
      from certifi import where
    File "certifi/__init__.py", line 1, in <module>
      from .core import contents, where
    File "certifi/core.py", line 12, in <module>
      from importlib.resources import read_text
    File "importlib/resources.py", line 11, in <module>
      from typing import Iterable, Iterator, Optional, Set, Union   # noqa: F401
    File "typing.py", line 1357, in <module>
      class Callable(extra=collections_abc.Callable, metaclass=CallableMeta):
    File "typing.py", line 1005, in __new__
      self._abc_registry = extra._abc_registry
  AttributeError: type object 'Callable' has no attribute '_abc_registry'
```

### PR DESCRIPTION
**Pull request by WoLpH**
For Python 3.5 and above the `typing` module is integrated in Python so it has no effect. But it seems to break `pip` in certain cases as shown in #144

### CODE REVIEW

**Comment by balloob** [comment]
`typing` is built into Python 3.5+. According to `setup.py`, it only supports Python 3.5+, so we should be able to remove it.

https://github.com/open-homeautomation/miflora/blob/ebda66d1f4ba71bc0b98f8383280e59302b40fc8/setup.py#L26-L29

Besides the reference in requirements.txt, it's also referenced in setup.py.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 bluepy==1.3.0
 pygatt==3.2.0
 btlewrap==0.0.9
-typing>=3,<4
+typing>=3,<4; python_version < 3.5
```
Comment (balloob) [inline requirements.txt:4]:
```suggestion
```

**Comment by balloob** [comment]
@open-homeautomation would you be able to take a look at this PR? Any project that uses modern Python versions has to `pip3 uninstall typing` after installing miflora, but that might not be clear for new users.

File: requirements.txt (hunk 1 above)
Comment (WoLpH) [reply requirements.txt:4]: Is there still a point to this? Even python 3.5 is going end-of-life within a few months and Python 3.4 has been end-of-life for about a year already

Comment (balloob) [reply requirements.txt:4]: my suggestion was to remove it completely instead of limiting it, since based on `setup.py`, Python 3.4 is not supported.

Comment (WoLpH) [reply requirements.txt:4]: oops, I misread :)

**Comment by bdraco** [comment]
@ChristianKuehnel Can you take a look?

**Comment by ChristianKuehnel** [comment]
Looks good to me!

Note: I also bumped the "btlewrap" dependency to 0.0.10, as it also depended in typing.

**Comment by balloob** [comment]
Could you also publish a new release?

**Comment by ChristianKuehnel** [comment]
Sorry, I do not have the premissions for that.

@open-homeautomation can you please create a new release?

**Comment by balloob** [comment]
@open-homeautomation would really like to see a new release.

**Comment by open-homeautomation** [comment]
Sorry, I can't find the time to maintain this. @ChristianKuehnel interested to take over this repository in the PIP releases?

**Comment by ChristianKuehnel** [comment]
Thx for the offer, but I'm not using miflora any more. Sorry 😞 . Maybe @ThomDietrich ?

I've moved to an [ESPHome](https://esphome.io/components/sensor/xiaomi_hhccjcy01.html) based solution. This proved to be much more reliable than Bluetooth on Linux.

**Pull request merged**